### ISSUE TITLE
 Creating an account should not require social login

### ISSUE THREAD

**Issue by ThHareau** [forem/forem#11603]
**Is your feature request related to a problem? Please describe.**

Creating an account should not require social login.

I want to create an account that would be using my professional email, however I'm using a personal github account as well as a personal twitter account.

**Describe the solution you'd like**

We should be able to create an account using an email only. 

**Describe alternatives you've considered**

I've managed to create an account this way:

* Login via Github
    * All my professional and private emails are collected (:disappointed:)
* From the settings, change the email to the professional account
* Setup a password from the setting
* Revoke github access

**Additional context**

**Comment by rhymes** [comment]
Hi @ThHareau, thanks for reporting this issue.

Regarding collection and data:

- when you link GitHub to DEV we only read the primary email: 

https://github.com/forem/forem/blob/1741e52e837b74eb982759edd4d5895b602beba0/app/services/authentication/providers/github.rb#L12

- when you set a password and revoke GitHub access the settings, we delete all the data connected to them, username included, thus we don't collect unneeded information, nor retain it

https://github.com/forem/forem/blob/1741e52e837b74eb982759edd4d5895b602beba0/app/controllers/users_controller.rb#L137

**Comment by ThHareau** [comment]
Hi @rhymes, 

I understand that you only read the primary email. In my case that's a concern because that's an email that I only want to share with important services (healthcare, gov stuff, ...), I want that any email I receive in this mailbox are important.

I'm also surprised that the token authorization asks for all emails. Can't you modify the token to only fetch one email? 
I'm also concerned that, even if your intentions are pure, something messes up and all email leaks.

And last but the more important for me: why not allowing email account creation if it's possible to bypass social login?

**Comment by thwaller** [comment]
I was referred to this topic by the "DEV Community Team" in response to a support request I had submitted. I wanted to register for the site and noticed there was three options for social login and signup, but no option to register without using social. I was informed that while prior uses could use the password to login, this is not offered to new users.

I am a bit surprised that a development community would not consider this from the aspect of users privacy concerns. The result is I am declining to register. Twitter is not a trusted platform on any level. I am not an Apple user, and with the closed source of Apple, they are also not a trusted platform. This leaves GitHub, and while they are seemingly the most trusted on the list, that does not remove the security risks and privacy violations.

I can understand that the use of social logins make the development process easier, but when it comes at a price of privacy and security to the user, this should not be forced on the user. Especially when the risks are not clearly disclosed to the user. Example, If I offered my users 2 methods to login, first is social and second is email/password + MFA... then offered a section titled "which do I choose".... It would say that social is very convenient to use, but comes with risk due to account exposure, tracking, etc. Email/pass requires another password to be kept, but the account security is in our hands, and we can be trusted due to ....list here. What will people choose?
